### DISCUSSION TITLE
Python Tabs vs Spaces

### DISCUSSION THREAD

**Discussion by ZorudaRinku** [community/community#7029] [Copilot Conversations]
Yes the age old question, seems github copilot is very fond of spaces rather than tabs. Is there any way to correct this? It'll put spaces in a tab only project often.

IDE: Pycharm

**Comment by jansorg** [comment]
Thanks for reporting! A few more details would help to fix this.

- Which file type were you editing?
- Could you post your code style settings for this file type? Here's an example for HTML
    ![Screenshot_20211030_220810](https://user-images.githubusercontent.com/11473063/139557136-29cb766f-ef99-43e3-ae91-0926d7dbb34a.png)

**Reply by ZorudaRinku** [reply]
File type is Python
Screenshot of code style:
![image](https://user-images.githubusercontent.com/13283879/140625540-6a822400-abc0-418f-b427-6e0b8839724e.png)

**Comment by linux-china** [comment]
Same to me to work with Makefile:  use spaces and not tab

![Snip20211030_133](https://user-images.githubusercontent.com/46711/139559566-00be3ebe-d59d-466f-be3c-012f5ae4840d.png)

![Snip20211030_134](https://user-images.githubusercontent.com/46711/139559568-b842b0ae-a6c4-41f9-b383-4f14c5a2d24e.png)

IDE: IntelliJ IDEA

**Comment by joshstrange** [comment]
Same issue here with TypeScript files. GitHub Copilot inserts code with space indention instead of the configured tabs.
![Screen Shot 2021-10-31 at 4 56 28 PM](https://user-images.githubusercontent.com/462796/139601018-dc747436-fb0e-407c-9000-8d0760904d8c.png)


IDE: IntelliJ IDEA

**Comment by noblecloud** [comment]
Same here.  PyCharm is configured to use tabs rather than spaces along with having my .editorconfig set for tabs.
<img width="229" alt="image" src="https://user-images.githubusercontent.com/6209052/141312474-9b27a24b-ffe0-4d07-8a4b-8b8938fcd553.png">

IDE: IntelliJ PyCharm

**Comment by LolnationCH** [comment]
This question/feedback could be renamed "Copilot doesn't used configured indent", since this seems to not only be not using tabs, but instead Copilot ignores your configured indentation preferences.

I have the same issue with Vs Code, configured to use 2 space as indent, Copilot keeps using 4 spaces (since most people use that).

It's not a deal breaker, since I can just run an auto-format command, but still a pain :(
